### PR TITLE
http_11_proxy: Make inner transport_socket config optional

### PR DESCRIPTION
**Pull request by tonya11en**
http_11_proxy: Make inner transport_socket config optional

Given that the top-level [Cluster.transport_socket](https://github.com/envoyproxy/envoy/blob/1a153166a6d1e9336ee8982d1a00ba98655c9d39/api/envoy/config/cluster/v3/cluster.proto#L1099) field is optional and defaults to plaintext, this should also be optional. gRPC is adding support for this transport socket, but they do not have a `raw_buffer` to explicitly configure. See https://github.com/grpc/proposal/pull/455#discussion_r1776143739 for additional context.

Risk Level: Low.
Testing: Existing tests.
Docs Changes: n/a
Release Notes: Done.

### CODE REVIEW

File: api/envoy/extensions/transport_sockets/http_11_proxy/v3/upstream_http_11_connect.proto [hunk 1]
```diff
@@ -35,5 +35,5 @@ option (udpa.annotations.file_status).package_version_status = ACTIVE;
 //
 message Http11ProxyUpstreamTransport {
   // The underlying transport socket being wrapped.
-  config.core.v3.TransportSocket transport_socket = 1 [(validate.rules).message = {required: true}];
+  config.core.v3.TransportSocket transport_socket = 1;
```
Comment (abeyad) [inline api/envoy/extensions/transport_sockets/http_11_proxy/v3/upstream_http_11_connect.proto:38]: can you say more about why the underlying socket is no longer required?

**Comment by abeyad** [comment]
Also, tests seem to be failing

/wait

**Comment by tonya11en** [comment]
> Also, tests seem to be failing
> 
> /wait

The failure isn't related to this PR:
```
================================================================================
INFO: Elapsed time: 2312.606s, Critical Path: 1217.31s
INFO: 26910 processes: 11952 remote cache hit, 11870 internal, 1 local, 2 processwrapper-sandbox, 3085 remote.
//test/integration:load_stats_integration_test                           FAILED in 377.3s
  /build/bazel_root/base/execroot/envoy/bazel-out/k8-dbg/testlogs/test/integration/load_stats_integration_test/test.log
```

File: api/envoy/extensions/transport_sockets/http_11_proxy/v3/upstream_http_11_connect.proto (hunk 1 above)
Comment (tonya11en) [reply api/envoy/extensions/transport_sockets/http_11_proxy/v3/upstream_http_11_connect.proto:38]: I thought the PR description covered the reasoning. If it's unclear, can you give me a pointer on what details you're looking for?

File: api/envoy/extensions/transport_sockets/http_11_proxy/v3/upstream_http_11_connect.proto [hunk 2]
```diff
@@ -35,5 +35,5 @@ option (udpa.annotations.file_status).package_version_status = ACTIVE;
 //
 message Http11ProxyUpstreamTransport {
   // The underlying transport socket being wrapped.
```
Comment (markdroth) [inline api/envoy/extensions/transport_sockets/http_11_proxy/v3/upstream_http_11_connect.proto:37]: Suggest adding a comment here that says something like "If not set, plaintext is used."

**Review by markdroth — COMMENTED** [review]
Thanks for doing this!

Does the Envoy implementation need to be changed to default to plaintext if this field is unset?

File: api/envoy/extensions/transport_sockets/http_11_proxy/v3/upstream_http_11_connect.proto (hunk 2 above)
Comment (tonya11en) [reply api/envoy/extensions/transport_sockets/http_11_proxy/v3/upstream_http_11_connect.proto:37]: Gotcha, will do.

**Comment by tonya11en** [comment]
> Thanks for doing this!
> 
> Does the Envoy implementation need to be changed to default to plaintext if this field is unset?

@markdroth no change was required to the implementation.

**Comment by markdroth** [comment]
/lgtm api

Thanks!

File: api/envoy/extensions/transport_sockets/http_11_proxy/v3/upstream_http_11_connect.proto (hunk 1 above)
Comment (abeyad) [reply api/envoy/extensions/transport_sockets/http_11_proxy/v3/upstream_http_11_connect.proto:38]: So sorry, I missed the PR description.  That explains it, thanks!

**Review by abeyad — DISMISSED** [review]
/lgtm api

**Comment by kyessenov** [comment]
Please fix the build:
```
envoy/extensions/transport_sockets/http_11_proxy/v3/upstream_http_11_connect.proto:8:1:Import "validate/validate.proto" is unused.```

**Review by markdroth — COMMENTED** [review]
/lgtm api

**Comment by kyessenov** [comment]
/retest
unrelated example flake

**Review by alyssawilk — COMMENTED** [review]
Change LGTM but please add a test of it being unset just to verify it works e2e
/wait

**Review by alyssawilk — APPROVED** [review]

**Review by abeyad — APPROVED** [review]
/lgtm api

**Comment by alyssawilk** [comment]
/retest 
(https://github.com/envoyproxy/envoy/issues/36494)

**Pull request merged**